### PR TITLE
Removed helper import from l4lb

### PR DESCRIPTION
**Pull request by 08volt**
issue https://github.com/kubernetes/ingress-gce/issues/1902

Removed helper dependencies from the cloud-provider

Originally posted by [Antonio Ojea](https://github.com/aojea) in https://github.com/kubernetes/ingress-gce/pull/1896#discussion_r1072096303

### CODE REVIEW

**Comment by sawsa307** [comment]
> /assign [Elina Akhmanova](https://github.com/code-elinka) [Cezary Zawadka](https://github.com/cezarygerard)

/assign @cezarygerard @code-elinka

**Comment by cezarygerard** [comment]
please run 

go mod tidy
go mod verify
go mod vendor


and see if this PR removes dependencies

File: pkg/l4lb/l4lbcommon.go [hunk 1]
```diff
@@ -98,10 +97,37 @@ func deleteAnnotation(ctx *context.ControllerContext, svc *v1.Service, annotatio
 	return patch.PatchServiceObjectMetadata(ctx.KubeClient.CoreV1(), svc, *newObjectMeta)
 }
 
+// LoadBalancerStatusEqual checks if load balancer status are equal
+func loadBalancerStatusEqual(l, r *v1.LoadBalancerStatus) bool {
```
Comment (aojea) [inline pkg/l4lb/l4lbcommon.go:101]: this is wrong, I fixed it the other day in kubernetes

https://github.com/kubernetes/kubernetes/pull/125225/files

**Comment by aojea** [comment]
> > please run
> > go mod tidy go mod verify go mod vendor
> > and see if this PR removes dependencies
> 
> Done, no dependences removed

```
o mod why k8s.io/cloud-provider
# k8s.io/cloud-provider
k8s.io/ingress-gce/cmd/glbc/app
k8s.io/cloud-provider
aojea@aojea:~/src/ingress-gce$ grep -r k8s.io/cloud-provider/ cmd/
aojea@aojea:~/src/ingress-gce$ grep -r k8s.io/cloud-provider cmd
cmd/glbc/app/clients.go:        cloudprovider "k8s.io/cloud-provider"
cmd/glbc/app/clients.go:        "k8s.io/cloud-provider-gcp/providers/gce"
cmd/glbc/main.go:       firewallcrclient "k8s.io/cloud-provider-gcp/crd/client/gcpfirewall/clientset/versioned"
cmd/glbc/main.go:       networkclient "k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned"
cmd/e2e-test/psc_test.go:       "k8s.io/cloud-provider-gcp/providers/gce"
cmd/e2e-test/upgrade/psc.go:    "k8s.io/cloud-provider-gcp/providers/gce"

```

**Pull request closed without merging**